donut/modules/calendar/static/script.js

```diff
@@ -0,0 +1,388 @@
+            
+    var startDay = parseInt(begin.substring(8, 10), 10);
+    var endDay = parseInt(end.substring(8, 10), 10);
+    var startMonth = parseInt(begin.substring(5, 7), 10);
+    var endMonth = parseInt(end.substring(5, 7), 10);
+    startDay = month > startMonth - 1 ? 1 : startDay;
+    endDay = month < endMonth - 1 ? daysOfMonth[month] : endDay;
+    for(var j = startDay; j <= endDay; j ++)
+    {
+      var divNode = document.createElement('div');
+      divNode.style.backgroundColor = colorLabel[curEvent.organizer.displayName];
+      divNode.className = 'events';
+      divNode.className += ' '+curEvent.organizer.displayName;
+      divNode.className += ' rounded-circle'; 
+      divNode.setAttribute("id", curEvent.id + j);
+      divNode.className += ' ' + curEvent.id;        
+               
+      var eventName = document.createElement('Label');
+      eventName.innerHTML = curEvent.summary;
+      divNode.appendChild(eventName);
+      if (month === d.getUTCMonth())
+      {
+        document.getElementById('day ' + j).appendChild(divNode);
+        $("#"+curEvent.id+j).on('click',{data: curEvent}, showEventInfo);
+      }
+      
+      if(!checked[0].checked){
+        divNode.style.display = "none"; 
+      }
+    }
+  }
+}
+
+// Delete event from our db and from the page
+function deleteEvent(event)
+{
+    $.ajax({
+      url: '/1/calendar/delete_event',
+      type: 'POST',
+      data:{
+        id:event.data.eventID,
+        tag:event.data.tag
+      },
+      success: function(data) {
+        if(data.deleted === '')
+        {
+            $('.'+event.data.eventID).remove();
+            $('#extraInfo_triangle').hide();
+            $('#extraInfo').hide();
+        }
+        else{
+            alert(data.deleted);
+        }
+      }
+    });
+}
+
+// Cache the events locally for an entire year. 
+function process_events(data, month, year)
+{
+ data.events = data.events || [];
+
+  for(var i = 0; i < data.events.length; i++)
+  {
+    var curEvent = data.events[i];
+    var beginStr = curEvent.start.dateTime || curEvent.start.date;
+    var beginMonth = parseInt(beginStr.substring(5, 7));
+    var endStr = curEvent.end.dateTime || curEvent.end.date;
+    var endMonth = parseInt(endStr.substring(5, 7));
+
+    // If the current year and the end year are not the same, then 
+    // for whatever ungodly reason the event spans a year
+    endMonth = (new Date(endStr).getFullYear() === year) ? endMonth : 12;
+
+    // Same but with beginning months
+    beginMonth =  (new Date(beginStr).getFullYear() === year) ? beginMonth : 1;
+    while(beginMonth <= endMonth)
+    {
+      events[beginMonth-1].push(curEvent);
+      beginMonth += 1;
+    }
+  }
+  renderEvents(month, year);
+}
+
+// Actually requests for the events from the server. 
+function getEvents(month, year){
+  // We don't always need to send a request. 
+  if (month === 11||month === 0 || $.isEmptyObject(events) || curYear != year)
```

Comment:
Why the special case for Jan/Dec? Shouldn't it depend on which direction you are going?

Reply:
Also this would be cleaner with an early `return` instead of wrapping the whole function in an `if`